### PR TITLE
docs: Improve documentation about how to debug metrics server

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ to deploy it as part of KEDA. Do the following:
     kubectl logs -l app=keda-metrics-apiserver -n keda -f
     ```
 
-## Debugging
+## Debugging Operator
```
Comment (tomkerkhove) [inline BUILD.md:125]: Nit - We might want to use h2 for "Debugging" and have h3 per operator/metric server

File: tests/scalers/cron/cron_test.go [hunk 2]
```diff
@@ -25,8 +25,8 @@ var (
 	scaledObjectName = fmt.Sprintf("%s-so", testName)
 
 	now   = time.Now().Local()
-	start = (now.Minute() + 1)
-	end   = (start + 1)
+	start = (now.Minute() + 1) % 60
```
Comment (tomkerkhove) [inline tests/scalers/cron/cron_test.go:28]: Intentional? Doesn't seem related to the docs

**Review by tomkerkhove — APPROVED** [review]

File: tests/scalers/cron/cron_test.go (hunk 2 above)
Comment (JorTurFer) [reply tests/scalers/cron/cron_test.go:28]: It is,  I can open another PR to be more clear. Running e2e test yesterday I found that the generated SO goes from 60 * * * * to 61 * * * * and test failed

Comment (tomkerkhove) [reply tests/scalers/cron/cron_test.go:28]: Let's split it in another PR if OK indeed

Comment (JorTurFer) [reply tests/scalers/cron/cron_test.go]: done! https://github.com/kedacore/keda/pull/3590

File: BUILD.md (hunk 1 above)
Comment (JorTurFer) [reply BUILD.md]: PTAL

**Comment by v-shenoy** [comment]
I think we should prefer and use links with `latest` instead of `2.8` (or a specific version). That way there's no need to update them later.

**Comment by JorTurFer** [comment]
> NIT: I think we should prefer and use links with `latest` instead of `2.8` (or a specific version). That way there's no need to update them later.

you are right!

**Review by tomkerkhove — APPROVED** [review]
